src/utils/nlpUtils/newEntitiesRuler.py

```diff
@@ -0,0 +1,57 @@
+'''
+Created on Sep 27, 2022
+
+@author: mandd
+'''
+
+import tagKeywordListReader as tklr
+
+def patternCreator(tagDict):
+  """
+    This method is designed to create patterns from the tags and the corresponding list of
+    keywords.
+    @ In, tagDict, dict, dictionary containing tags and keywords. This dictionary is generated
+                         by keyWordListGenerator() located in tagKeywordListReader.py from the
+                         file nlp/data/tag_keywords_lists.xlsx
+    @ Out, patterns, list, list of patterns: {"label": label_ID, "pattern": keyword}
+  """
+  patterns = []
+  for tag in tagDict:
+    for elem in tagDict[tag]:
+      listElem ={"label": tag, "pattern": elem}
+      patterns.append(listElem)
+  return patterns
+
+def unitPatternCreator(measureDict):
+  """
```

Comment:
I think we could combine these two methods with nlp_utils.generatePatternList method.

Reply:
I think I need your help here

Reply:
This can be resolved in future development.